pipelines/pipedrive/custom_fields_munger.py

```diff
@@ -0,0 +1,56 @@
+from dlt.common.normalizers.names.snake_case import normalize_column_name
+from typing import Dict, List, Generator, Optional
+
+import dlt
+
+
+def munge_push_func(data: List, endpoint: str) -> List:
+    """
+    Specific function to perform data munging and push changes to custom fields' mapping stored in dlt's state
+    The endpoint must be an entity fields' endpoint
+    """
+    custom_fields_mapping = dlt.state().setdefault('custom_fields_mapping', {})
+    for data_item in data:
+        if all([data_item.get('edit_flag'), data_item.get('name'), data_item.get('key')]):
+            if custom_fields_mapping.get(endpoint):
+                if data_item['key'] not in custom_fields_mapping[endpoint]:
+                    data_item_mapping = _normalize_map(data_item)
+                    custom_fields_mapping[endpoint].update(data_item_mapping)
+            else:
+                data_item_mapping = _normalize_map(data_item)
+                custom_fields_mapping[endpoint] = data_item_mapping
+            data_item['key'] = custom_fields_mapping[endpoint][data_item['key']]['normalized_name']
```

Comment:
why we need this line?

Reply:
Same as before.
First of all I search on/update/add values on dlt's state, and then I end up updating data with dlt's state values.

Reply:
OK I get it and this is good! please move the explanations above to the comments

Reply:
Sure!